pandapower/results.py

```diff
@@ -79,9 +79,19 @@ def init_element(net, element, suffix=None):
     res_element, res_empty_element = get_result_tables(element, suffix)
     index = net[element].index
     if len(index):
-        # init empty dataframe
-        res_columns = net[res_empty_element].columns
-        net[res_element] = pd.DataFrame(np.nan, index=index, columns=res_columns, dtype='float')
+        if hasattr(net, '_options'):
+            if net._options['mode'] is 'opf' and element is 'xward':
+                    # In case of an OPF in a net containing xwards, keep vm_internal_pu values since they need to be
+                    # constrained
+                    net.res_xward.loc[:, net.res_xward.columns.drop('vm_internal_pu')] = np.nan
+            else:
+                # init empty dataframe
+                res_columns = net[res_empty_element].columns
+                net[res_element] = pd.DataFrame(np.nan, index=index, columns=res_columns, dtype='float')
+        else:
+            # init empty dataframe
+            res_columns = net[res_empty_element].columns
+            net[res_element] = pd.DataFrame(np.nan, index=index, columns=res_columns, dtype='float')
```

Comment:
Is there a way to avoid code duplication here?

Reply:
Not to the best of my knowledge

Reply:
Isn't it the exactly same code? Why not jsut put it in a function?